c10/core/ScalarType.h

```diff
@@ -51,17 +56,41 @@ _(int,Int,i)       \
 _(int64_t,Long,i)  \
 _(at::Half,Half,d) \
 _(float,Float,d)   \
+_(double,Double,d) \
+_(c10::qint8,QInt8,i)
+
+#define AT_FORALL_SCALAR_TYPES_EXCEPT_QINT(_) \
```

Comment:
these macros are getting unwieldy (not your fault).  I'll try to figure out something here.